lib/libc/Kconfig

```diff
@@ -19,7 +19,7 @@ config SUPPORT_MINIMAL_LIBC
 config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
-	depends on "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "arcmwdt"
+	depends on "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "arcmwdt" || "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "xcc" || "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "xcc-clang"
```

Comment:
I think this should be changed to:
depends on "$(ZEPHYR_TOOLCHAIN_VARIANT)" == "zephyr".

Reply:
> We should limit picolibc to zephyr toolchain for the release

Why? Picolibc should work with many GCC-based toolchain variants we currently support including, but not necessarily limited to, `cross-compile`, `espressif` and `gnuarmemb`.

Reply:
Maybe add those that are supported instead of declaring this supported on all and deal with issues on toolchains that were not verified.

Reply:
Seems like a reasonable plan for this release. Shall we plan on adding more toolchains as 'likely supported' after the release (like xcc-clang)?